src/drivers/serac.cpp

```diff
@@ -158,16 +148,42 @@ int main(int argc, char* argv[])
   solid_solver.setDisplacementBCs(ess_bdr, disp_coef);
   solid_solver.setTractionBCs(trac_bdr, traction_coef);
 
+  // neo-Hookean material parameters
+  double mu, K;
+  inlet->get("nonlinear_solid/mu", mu);  // has default value
+  inlet->get("nonlinear_solid/K", K);    // has default value
+
   // Set the material parameters
   solid_solver.setHyperelasticMaterialParameters(mu, K);
 
-  // Set the linear solver parameters
-  if (gmres_solver == true) {
+  // Solver parameters
+  serac::NonlinearSolverParameters nonlin_params;
+  inlet->get("nonlinear_solid/solver/nonlinear/rel_tol", nonlin_params.rel_tol);          // has default value
```

Comment:
This kind of shows a needed functionality in Inlet to get a Table then find child values from that given table.  Don't know how common a very nested value would be but I'm going to create an issue on this for future work.

Reply:
I think it would be extremely common.

Reply:
do you think it's possible to support a syntax like

```cpp
auto nonlin_params = inlet->get<serac::NonlinearSolverParameters>("nonlinear_solid/solver/nonlinear");
```

it's similar to what https://github.com/nlohmann/json does, and is reminiscent of `std::get<T>` in the standard library.

(the user provides a definition of the conversion to a specified type somewhere)

Reply:
The json-analogous way of doing this is defining two free functions `void to_inlet(const T& t, sidre::group& g)` and `void from_inlet(const sidre::group& g, T& t)`, but this could also be done with constructors and conversion operators

Reply:
I'm not familiar with the way inlet is implemented, but the approach Josh and I are describing makes it really easy to define arbitrarily nested data types too.

Reply:
I am open to suggestions.  The next big push on inlet is to better support nested Tables and arrays of Tables.